packages/documentation/gatsby-node.js

```diff
@@ -8,69 +6,226 @@
+                headings {
+                  value
+                }
+                html
+              }
+              gitRemote {
+                ref
+                webLink
+              }
+              internal {
+                mediaType
+              }
+              name
+              relativeDirectory
+              relativePath
+              sourceInstanceName
             }
-            name
-            relativeDirectory
-            relativePath
-            sourceInstanceName
           }
         }
       }
-    }
-  `);
+    `
+  );
 
-  result.data.allFile.edges.forEach(({ node }) => {
-    const {
-      webLink: repositoryUrl,
-      ref: branch,
-    } = node.gitRemote;
+  if (result.errors) {
+    reporter.panicOnBuild(
+      'Error querying for va.gov-team md files',
+      new Error(result.errors),
+    );
+  }
 
+  result.data.allFile.edges.forEach(({ node }) => {
     const { headings, html } = node.childMarkdownRemark;
 
+    // Don't create pages for empty files.
+    if (!html) return;
+
+    const {
+      gitRemote: {
+        webLink: repositoryUrl,
+        ref: branch,
+      },
+      name,
+      relativeDirectory,
+      relativePath,
+      sourceInstanceName: source,
+    } = node;
+
+    // Use the top level heading as the title.
     const title = headings && headings[0] && headings[0].value;
 
-    // Link to GitHub document.
-    const sourceUrl = `${repositoryUrl}/blob/${branch}/${node.relativePath}`;
+    // Build the full link to the document.
+    const sourceUrl = `${repositoryUrl}/blob/${branch}/${relativePath}`;
 
     createPage({
-      path: `${node.relativeDirectory}/${node.name}`,
+      path: `${relativeDirectory}/${name}`,
       component: path.resolve('./src/templates/page.js'),
       context: {
         html,
+        source,
         sourceUrl,
         title,
       },
-    })
+    });
+  });
+};
+
+/**
+ * Derive and include the source URL on automatically generated pages,
+ * specifically from src/pages.
+ *
+ * External pages, such as those pulled from va.gov-team or formation-react,
+ * should already have sources and also source URLs associated with them.
+ */
+const setSourceUrl = ({ page, actions, reporter }) => {
+  if (page.context.sourceUrl) return;
+
+  // For debugging, log if an externally sourced page is missing the URL.
+  if (page.context.source) {
+    reporter.warn(`No source URL found for external page: ${page}`);
+    return;
+  }
+
+  // Check if this page is sourced from src/pages.
+  const { createPage, deletePage } = actions;
+  const match = page.componentPath && page.componentPath.match(
+    `/packages/documentation/src/pages/.+`
+  );
+
+  // Avoid setting source URLs that we can't confidently determine.
+  if (!match) return;
+
+  // Build the source URL once we've determined that the page is from src/pages.
+  const [relativePath] = match;
+  const sourceUrl = `${GITHUB_FILE_BASE_URL}${relativePath}`;
+
+  deletePage(page);
+
+  createPage({
```

Comment:
Does creating the page after deleting it allow you to effectively add `context` to the page without creating a duplicate page?

Reply:
Yup, that appears to be the [official way of adding context](https://www.gatsbyjs.com/docs/creating-and-modifying-pages/#pass-context-to-pages).

Reply:
Cool. Thanks for linking to the docs.